pkg/operator/cassandra/sidecar/sidecar.go

```diff
@@ -0,0 +1,166 @@
+package sidecar
+
+import (
+	"fmt"
+	"github.com/coreos/pkg/capnslog"
+	rookClientset "github.com/rook/rook/pkg/client/clientset/versioned"
+	"github.com/rook/rook/pkg/operator/cassandra/constants"
+	"github.com/rook/rook/pkg/operator/cassandra/nodetool"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/workqueue"
+	"net/http"
+	"net/url"
+	"time"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "sidecar")
+
+// MemberController encapsulates all the tools the sidecar needs to
+// talk to the Kubernetes API
+type MemberController struct {
+	name, namespace, ip       string
+	cluster, datacenter, rack string
+	kubeClient                kubernetes.Interface
+	rookClient                rookClientset.Interface
+
+	nodetool nodetool.Interface
+
+	queue workqueue.RateLimitingInterface
+}
+
+// New return a new MemberController
+func New(
+	name, namespace string,
+	kubeClient kubernetes.Interface,
+	rookClient rookClientset.Interface,
+) (*MemberController, error) {
+
+	// Retrieve the service that we use as a static IP
+	idService, err := kubeClient.CoreV1().Services(namespace).Get(name, metav1.GetOptions{})
+	for err != nil || idService.Spec.ClusterIP == "" {
+		logger.Infof("Something went wrong trying to get Member Service %s", err.Error())
```

Comment:
should we have a break condition from this loop so when something is wrong we don't sit in it forever?

Reply:
If we can't find the seeds we can't do anything, so no point in breaking.